src/lib/openjp2/cio.c

```diff
@@ -264,6 +272,11 @@ void OPJ_CALLCONV opj_stream_set_user_data(opj_stream_t* p_stream,
     if (!l_stream) {
         return;
     }
+    /* CPB: if called after default chunk of memory allocated */
+    if (l_stream->m_user_data) {
+        p_function(l_stream->m_user_data);
```

Comment:
What is the purpose of p_function()?

Reply:
This frees up memory. The l_stream->m_user_data is assumed to be NULL, if not, free up whatever was left over using the appropriate function (as passed in through parameter).